### PR TITLE
Clean up AMP test folder

### PR DESCRIPTION
**Pull request by johannkm**
Name the folder auto_materialize_tests instead of asset_reconciliation. Put the scenarios in a sub folder

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15573** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15573" target="_blank">Graphite</a> 
    * **PR #15574** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15574" target="_blank">Graphite</a> 
      * **PR #15575** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15575" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15575?utm_source=stack-comment).

**Comment by sryza** [comment]
going to defer to Owen on this one, but I highly approve of cleaning up the AMP test folder

File: python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/base_scenario.py [hunk 1]
```diff
@@ -98,10 +98,14 @@ class AssetReconciliationScenario(NamedTuple):
     def _get_code_location_origin(
         self, scenario_name, location_name=None
     ) -> InProcessCodeLocationOrigin:
+        """scenarios.py puts all the scenarios in its namespaace under different 'hacky_daemon_repo_...' names.
```
Comment (OwenKephart) [inline python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/base_scenario.py:101]: nit: namespaaaaace

**Review by OwenKephart — APPROVED** [review]
great

**Pull request merged**